distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/ActionsBuilder.java

```diff
@@ -0,0 +1,53 @@
+package com.gruppo1.distributednetworkmanager;
+
+import com.dezen.riccardo.networkmanager.Resource;
+import com.dezen.riccardo.smshandler.Peer;
+
+/**
+ *
+ * @param <T> the real class that describes an Action of you custom network
+ * @param <P> a real type of peer that can be contacted
+ * @param <R> a resource type
+ * @param <N> a Node of the network
+ */
+public interface ActionsBuilder<T, P extends Peer, R extends Resource, N extends Node> {
+
+    T buildPing(P peer);
+
+
+    /**
+     * Hypothesis: this type of responses (ANSW) is built starting from the request (mainly to retrieve the action ID/number)
+     *
+     * QUESTION: is it good?
+     */
+    T buildPingAnsw(T request);
+
+
+    T buildInvite(P peer);
+
+    T buildInviteAnsw(T request, boolean accept);
+
+
+
+    T[] buildStore(P peer, P peerToStore);
+
+    T buildStore(P peer, R resource);
+
+    T[] buildStoreAnsw(T request, P peer);
+
+    T buildStoreAnsw(T request, boolean confirmStore);
+
+
+
+    T buildFindNode(P peer, N node);
+
+    T[] buildFindNodeAnsw(T request, N[] nodes);
```

Comment:
Why does find Node answer take in Nodes but other answer Peers? We should decide on one, the fact our PeerNodes extend both Peer and Node is a coincidence we shouldn't consider in the interface

Reply:
Yeah, buildFindNodeAnsw should respond with P[] peers